gusto/equations/shallow_water_equations.py

```diff
@@ -272,7 +232,245 @@ def __init__(self, domain, parameters, fexpr=None, bexpr=None,
+            linearisation_map (func, optional): a function specifying which
+                terms in the equation set to linearise. If None is specified
+                then no terms are linearised. Defaults to the string 'default',
+                in which case the linearisation includes both time derivatives,
+                the 'D' transport term and the pressure gradient term.
+            u_transport_option (str, optional): specifies the transport term
+                used for the velocity equation. Supported options are:
+                'vector_invariant_form', 'vector_advection_form', and
+                'circulation_form'.
+                Defaults to 'vector_invariant_form'.
+            no_normal_flow_bc_ids (list, optional): a list of IDs of domain
+                boundaries at which no normal flow will be enforced. Defaults to
+                None.
+            active_tracers (list, optional): a list of `ActiveTracer` objects
+                that encode the metadata for any active tracers to be included
+                in the equations. Defaults to None.
+        """
+
+        self.equivalent_buoyancy = equivalent_buoyancy
+        b_name = 'b_e' if equivalent_buoyancy else 'b'
+        self.b_name = b_name
+
+        # Note: we do not pass bexpr to the parent class because we
+        # deal with topography terms here later
+        super().__init__(domain, parameters,
+                         fexpr=fexpr, bexpr=None,
+                         space_names=space_names,
+                         u_transport_option=u_transport_option,
+                         linearisation_map=linearisation_map,
+                         no_normal_flow_bc_ids=no_normal_flow_bc_ids,
+                         active_tracers=active_tracers)
+
+        self.parameters = parameters
+        g = parameters.g
+        H = parameters.H
+
+        w = self.tests[0]
+        gamma = self.tests[2]
+        u, D, b = split(self.X)[0:3]
+        n = FacetNormal(domain.mesh)
+        topog = self.prescribed_fields('topography', domain.spaces('DG')).interpolate(bexpr) if bexpr else None
+        if equivalent_buoyancy:
+            gamma_qt = self.tests[3]
+            qt = split(self.X)[3]
+
+        # -------------------------------------------------------------------- #
+        # Add pressure gradient-like terms to residual
+        # -------------------------------------------------------------------- #
+        # drop usual pressure gradient term
+        residual = self.residual.label_map(
+            lambda t: t.has_label(pressure_gradient),
+            drop)
+
+        # add (moist) thermal source terms not involving topography -
+        # label these as the equivalent pressure gradient term
+        if equivalent_buoyancy:
+            q0 = parameters.q0
+            nu = parameters.nu
+            beta2 = parameters.beta2
+            qsat_expr = compute_saturation(q0, nu, H, g, D, b_name, topog)
```

Comment:
I think we want `b` here as the argument rather than `b_name`.

Reply:
yes, good spot!